libp2p/Host.cpp

```diff
@@ -989,13 +997,35 @@ bool Host::peerSlotsAvailable(Host::PeerSlotType _type /*= Ingress*/)
     return peerCount() + m_pendingPeerConns.size() < peerSlots(_type);
 }
 
-KeyPair Host::networkAlias(bytesConstRef _b)
+std::pair<Secret, ENR> Host::restoreENR(bytesConstRef _b, NetworkConfig const& _netConfig)
 {
     RLP r(_b);
+    Secret secret;
     if (r.itemCount() == 3 && r[0].isInt() && r[0].toInt<unsigned>() >= 3)
-        return KeyPair(Secret(r[1].toBytes()));
+    {
+        if (r[1].isList())
+        {
+            secret = Secret{r[1][0].toBytes()};
+            auto enrRlp = r[1][1];
+
+            return make_pair(secret, parseV4ENR(enrRlp));
+        }
+
+        // Support for older format without ENR
+        secret = Secret{Secret(r[1].toBytes())};
+    }
     else
-        return KeyPair::create();
+    {
+        // no private key found, create new one
+        secret = KeyPair::create().secret();
+    }
+
+    // TODO(gumb0): update ENR in case new address given in config
```

Comment:
Can we add an issue ID here for tracking purposes?

Reply:
I've created an issue covering all ENR Extension tasks https://github.com/ethereum/aleth/issues/5551, will link it here